src/analysis/SmartDetectorsAnalysis/SmartDetectorRunner.cs

```diff
@@ -153,7 +153,13 @@ private static string GetResolutionStateKey(string alertCorrelationHash)
             try
             {
                 alerts = await smartDetector.AnalyzeResourcesAsync(analysisRequest, detectorTracer, cancellationToken);
-                this.tracer.TraceInformation($"Completed running Smart Detector ID {smartDetectorManifest.Id}, Name {smartDetectorManifest.Name}, returning {alerts.Count} alerts");
+                this.tracer.TraceInformation(
+                    $"Completed running Smart Detector ID {smartDetectorManifest.Id}, Name {smartDetectorManifest.Name}, returning {alerts.Count} alerts");
+            }
+            catch (DetectorDataNotReadyException ddnre)
+            {
+                this.tracer.TraceWarning($"Smart Detector data is not ready yet, aborting analysis: {ddnre.Message}");
+                return new List<ContractsAlert>();
```

Comment:
Is that temporary? otherwise how will you know to return it to the queue

Reply:
this is not supported in the Monitoring Appliance - it's an optional feature than may or may not be implemented by the Runtime Environment

Reply:
Can we add some kind of a flag to analysis request to let the user know if the runtime environment supports this?

---
In reply to: [243836908](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/38#discussion_r243836908) [](ancestors = 243836908)

Reply:
i would rather not - the flag's value should not change the detector's decision in any case

---
In reply to: [243896042](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/38#discussion_r243896042) [](ancestors = 243896042,243836908)